transformations/summarization_transformation/transformation.py

```diff
@@ -0,0 +1,212 @@
+                    objs.extend(potentialNewObjs)
+        return objs
+
+    def getObjFromXComp(self, deps):
+        for dep in deps:
+            if dep.pos_ == "VERB" and (dep.dep_ == "xcomp" or dep.dep_ == "pcomp"):
+                v = dep
+                rights = list(v.rights)
+                objs = [tok for tok in rights if tok.dep_ in self.dicts['OBJECTS']]
+                objs.extend(self.getObjsFromPrepositions(rights))
+                if len(objs) > 0:
+                    return v, objs
+        return None, None
+
+    def getObjsFromConjunctions(self, objs):
+        """
+        Similar with finding subjects with conjunctions
+        """
+        moreObjs = []
+        for obj in objs:
+            # rights is a generator
+            rights = list(obj.rights)
+            rightDeps = {tok.lower_ for tok in rights}
+            if "and" in rightDeps:
+                moreObjs.extend([tok for tok in rights if tok.dep_ in self.dict['OBJECTS'] or tok.pos_ == "NOUN"])
+                if len(moreObjs) > 0:
+                    moreObjs.extend(self.getObjsFromConjunctions(moreObjs))
+        return moreObjs
+
+    def getVerbsFromConjunctions(self, verbs):
+
+        """
+        Sometimes the verbs come with pair with conjunctions, or two clauses connected with conjunctions
+        """
+        moreVerbs = []
+        for verb in verbs:
+            rightDeps = {tok.lower_ for tok in verb.rights}
+            if "and" in rightDeps:
+                moreVerbs.extend([tok for tok in verb.rights if tok.pos_ == "VERB"])
+                if len(moreVerbs) > 0:
+                    moreVerbs.extend(self.getVerbsFromConjunctions(moreVerbs))
+        return moreVerbs
+
+    def findSubs(self, tok):
+        """
+        Based on the verbs, search backwards to find the subjects
+        """
+        head = tok.head
+        while head.pos_ != "VERB" and head.pos_ != "NOUN" and head.head != head:
+            head = head.head
+        if head.pos_ == "VERB":
+            subs = [tok for tok in head.lefts if tok.dep_ == "SUB"]
+            if len(subs) > 0:
+                verbNegated = self.isNegated(head)
+                subs.extend(self.getSubsFromConjunctions(subs))
+                return subs, verbNegated
+            elif head.head != head:
+                return self.findSubs(head)
+        elif head.pos_ == "NOUN":
+            return [head], self.isNegated(tok)
+        return [], False
+
+    def getAllSubs(self, v):
+        verbNegated = self.isNegated(v)
+        subs = [tok for tok in v.lefts if tok.dep_ in self.dicts['SUBJECTS'] and tok.pos_ != "DET"]
+        if len(subs) > 0:
+            subs.extend(self.getSubsFromConjunctions(subs))
+        else:
+            foundSubs, verbNegated = self.findSubs(v)
+            subs.extend(foundSubs)
+        return subs, verbNegated
+
+    def getAllObjs(self, v):
+
```

Comment:
Extra space?

Reply:
Thank you for the suggestions! I have modified them. We really appreciate your helpful inputs!